src/libs/OptionsListUtils.js

```diff
@@ -185,13 +185,10 @@ function getOptions(reports, personalDetails, draftComments, activeReportID, {
 
         // Skip this entry if it has no comments and is not the active report. We will only show reports from
         // people we have sent or received at least one message with.
-        const hasNoComments = report.lastMessageTimestamp === 0;
-        const shouldFilterReport = !showReportsWithNoComments && hasNoComments
-            && report.reportID !== activeReportID && !report.isPinned;
-        if (shouldFilterReport) {
-            return;
-        }
-        if (hideReadReports && report.unreadActionCount === 0 && !report.isPinned) {
+        const shouldFilterReportIfEmpty = !showReportsWithNoComments && report.lastMessageTimestamp === 0;
+        const shouldFilterReportIfRead = hideReadReports && report.unreadActionCount === 0;
+        const shouldFilterReport = shouldFilterReportIfEmpty || shouldFilterReportIfRead;
+        if (report.reportID !== activeReportID && !report.isPinned && shouldFilterReport) {
```

Comment:
The logic here looks good and the tests pass! However, with this new logic it seems the comment above the const declarations is out of date (it doesn't take into account pinned reports or gsd mode). IMO the const names and logic read easily, so you can just remove that comment.

Reply:
I have updated the comments. Let me know if they look fine. Let's keep some sorts of comments as in the future we could need more cases.